### ISSUE TITLE
Java14 enhanced instaceof check validation: ConstantNameCheck

### ISSUE THREAD

**Issue by nmancus1** [checkstyle/checkstyle#8404]
Child of #8391 
Check documentation: https://checkstyle.sourceforge.io/config_naming.html#ConstantName

From documentation:
_Checks that constant names conform to a specified pattern. A constant is a static and final field or an interface/annotation field, except serialVersionUID and serialPersistentFields._

Since a pattern variable is not a field, nor is it static, this check should not apply, but we should make a UT to ensure that we avoid false positives since this check looks at variable definitions.  After #8401 , this check works as intended:

```
➜  instanceof-pattern-var-def /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 TestClass.java
Note: TestClass.java uses preview language features.
Note: Recompile with -Xlint:preview for details.
➜  instanceof-pattern-var-def cat config.xml       
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
      <module name="ConstantName">
          <property name="format"
                    value="^log(ger)?$|^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$"/>
      </module>
  </module>
</module>
➜  instanceof-pattern-var-def cat TestClass.java   
public class TestClass {
    Integer magicNumber = 42;
    {
       if (magicNumber instanceof Integer x) { 
           System.out.println(x);
       } 
    }
}

➜  instanceof-pattern-var-def java $RUN_LOCALE -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.35-SNAPSHOT-all.jar -c config.xml TestClass.java
Starting audit...
Audit done.

```

**Comment by romani** [comment]
This Check do not care about usage of constant, it care about `Integer magicNumber = 42;` only. No updates are required.
